### PR TITLE
Implement parsing of struct decls

### LINKED ISSUE #12: Implement parsing of structs

### PR DESCRIPTION
**Pull request by rightfold**
Fixes #12

### CODE REVIEW

File: src-lib/Language/Mill/AST.hs [hunk 1]
```diff
@@ -26,8 +26,12 @@ data Decl
     | SubDecl String ParameterList Type Expr
     | ForeignSubDecl ForeignSource CallingConvention String ParameterList Type
     | AliasDecl String Type
+    | StructDecl String [Field]
     deriving (Eq, Show)
 
+data Field = Field String Type
+             deriving (Eq, Show)
```
Comment (vendethiel) [inline src-lib/Language/Mill/AST.hs]: this is basically our `Parameter`

File: src-lib/Language/Mill/Parse.hs [hunk 2]
```diff
@@ -84,6 +84,18 @@ aliasDecl = do
...
+structDecl :: Parser Decl
+structDecl = do
+    structKeyword
+    name <- identifier
+    openingBrace
+    fields <- many field
+    closingBrace
+    return $ StructDecl name fields
+    where
+        field :: Parser Field
```
Comment (vendethiel) [inline src-lib/Language/Mill/Parse.hs]: seems weird considering "parameter" is not in a `where`?

Comment (rightfold) [inline src-lib/Language/Mill/Parse.hs]: Feel free to refactor :p

Comment (vendethiel) [inline src-lib/Language/Mill/Parse.hs]: which way around?

Comment (rightfold) [inline src-lib/Language/Mill/Parse.hs]: towards `where`.

Comment (vendethiel) [inline src-lib/Language/Mill/Parse.hs]: means `parameter` is not testable by itself, though.

**Pull request merged**